What does lumpyexpress do? Returns the structural variant data but is that relevant?

Freebayes does something to get the SNP and INDEL quality value by running qv_estimate after

FRC gives you the features listed in the table that are also used to make the ideogram plots. But the FRC stats are used for the line graph. What are the FRC stats and how can they be summarized?


#5/31/22

Can I get a heterozygosity score or something from the merqury output of hairy vetch? Something to compare to common vetch. MAybe even get reads from all assemblies to get kmer plots for all?


Definitely check lentil and grasspea circos plot. But also check all other plots. What is expected of species this evolutionarily close? How do I discuss the circos plot findings in paper?

-Lentil and grasspea had one alignment in previous run
-Grasspea and red clover has one alignment
-Cowpea and common bean have multiple that are just really nice. Similar chromosome regions and several line up to same chromosome across species
-MedTr and red clover has two alignments

Find scientific names of all plants. Look into how they are related


###quickly run circos with 150000 and 200000 just for kicks

```bash
module load python_3/3.6.6 miniconda/3.6 

conda activate /project/forage_assemblies/tfuller/Themis-ASM/.snakemake/conda/3626987c/

python scripts/pafCircosPlotter.py -p mapped/mapVvillosa_CommonVetch.paf -r fastas/Vvillosa.fa -q fastas/CommonVetch.fa -o circos_150000 -c 10 -m 150000

python scripts/pafCircosPlotter.py -p mapped/mapVvillosa_CommonVetch.paf -r fastas/Vvillosa.fa -q fastas/CommonVetch.fa -o circos_200000 -c 10 -m 200000
```

both showed some things. 200000 only had two lines but 150000 had a few lines corresponding to denser linkages of the 100000 plot

###Run themis with circos plot minimum at 100000

```bash
conda activate 



sbatch -N 1 -n 2 --mem=25000 -p priority -q msn snakemake -s /project/forage_assemblies/tfuller/Themis-ASM/themisSnakefile --jobs 50 -p --use-conda --cluster-config /project/forage_assemblies/tfuller/Themis-ASM/cluster.json --cluster "sbatch --nodes={cluster.nodes} --ntasks-per-node={cluster.ntasks-per-node} --mem={cluster.mem} --partition={cluster.partition} -q {cluster.qos} -o {cluster.stdout}" --touch

job id 7757779

sbatch -N 1 -n 2 --mem=25000 -p priority -q msn snakemake -s /project/forage_assemblies/tfuller/Themis-ASM/themisSnakefile --jobs 50 -p --use-conda --cluster-config /project/forage_assemblies/tfuller/Themis-ASM/cluster.json --cluster "sbatch --nodes={cluster.nodes} --ntasks-per-node={cluster.ntasks-per-node} --mem={cluster.mem} -t 2-0 --partition={cluster.partition} -q {cluster.qos} -o {cluster.stdout}" --dryrun

job id 7757782

sbatch -N 1 -n 2 --mem=50000 -p priority -q msn --wrap="python themisASM.py  \
	-a /project/forage_assemblies/tfuller/Themis-ASM/vetch_phase.assembly.fasta  \
	-a /project/forage_assemblies/tfuller/Themis-ASM/Lens_culinaris_2.0.fasta  \
	-a /project/forage_assemblies/tfuller/Themis-ASM/Ler.1DRT.fasta \
	-a /project/forage_assemblies/tfuller/Themis-ASM/Vunguiculata_469_v1.0.fa \
	-a /project/forage_assemblies/tfuller/Themis-ASM/Pvulgaris_442_v2.0.fa \
	-a /project/forage_assemblies/tfuller/Themis-ASM/Pisum_sativum_v1a.fa \
	-a /project/forage_assemblies/tfuller/Themis-ASM/MtrunA17r5.0-20161119-ANR.genome.fasta \
	-a /project/forage_assemblies/tfuller/Themis-ASM/ARS_DFRC_1.1.fasta \
	-a /project/forage_assemblies/tfuller/Themis-ASM/Vicia_sativa_reorg.fa \
	-n Vvillosa \
	-n Lentil \
	-n GrassPea \
	-n CowPea \
	-n CommonBean \
	-n Pea \
	-n MedTr \
	-n RedClover \
	-n CommonVetch \
	-b eudicots_odb10 \
	-f /project/forage_assemblies/tfuller/Themis-ASM/HV30Vetch_S1_L001_R1_001.fastq.gz,/project/forage_assemblies/tfuller/Themis-ASM/HV30Vetch_S1_L001_R2_001.fastq.gz \
	-f /project/forage_assemblies/tfuller/Themis-ASM/HV30Vetch_S1_L002_R1_001.fastq.gz,/project/forage_assemblies/tfuller/Themis-ASM/HV30Vetch_S1_L002_R2_001.fastq.gz \
	-f /project/forage_assemblies/tfuller/Themis-ASM/HV30Vetch_S1_L003_R1_001.fastq.gz,/project/forage_assemblies/tfuller/Themis-ASM/HV30Vetch_S1_L003_R2_001.fastq.gz \
	-f /project/forage_assemblies/tfuller/Themis-ASM/HV30Vetch_S1_L004_R1_001.fastq.gz,/project/forage_assemblies/tfuller/Themis-ASM/HV30Vetch_S1_L004_R2_001.fastq.gz \
	-s lane1 \
	-s lane2 \
	-s lane3 \
	-s lane4 \
	-j 50 \
	-c 'sbatch --nodes={cluster.nodes} --ntasks-per-node={cluster.ntasks-per-node} --mem={cluster.mem} --partition={cluster.partition} -q {cluster.qos} -o {cluster.stdout}' \
	-r"

job id 7757783
```

-Lentil and grasspea have multiple lines going matching chromosome numbers. make thick looking paint stripes
-Grasspea and red clover still only has the one alignment
-Cowpea and common bean have even more really thick lines that are just really nice. Similar chromosome regions and several line up to same chromosome across species
-MedTr and red clover interestingly has no alignments. Likely because the MedTr chromosomes are so small that the 5% was smaller than 100000bp.